### ISSUE TITLE
Unicode characters with bytes greater than 127 cause crash

### ISSUE THREAD

**Issue by collinhundley** [vapor/mysql#25]
The `Char` type used in `Bind.swift` is an `Int8`, which only supports values up to 127. So, attempting to store any Unicode characters higher than this causes a crash. The crash occurs at line 71 in `Bind.swift`, while attempting to initialize a `Char` from a single byte if the byte's value is over 127.

Is there any reason `Char` can't be something bigger? `Int32` should support all Unicode characters that I'm aware of.

**Comment by tannernelson** [comment]
If you could submit a test case where this fails, that would help a ton.

Otherwise I will investigate this on the next update.

**Comment by collinhundley** [comment]
Here's how to reproduce the issue:

```swift
let mysql = try! MySQL.Database(...)
let myString = "È" // Unicode #200
try! mysql.execute("INSERT INTO mytable (myfield) VALUES (?)", [.string(myString)], nil)
```

This crashes at the following line in `Bind.swift`:

```swift
buffer[i] = Char(byte)
```

Because it's attempting to cast a decimal value of 200 into an `Int8`.

**Comment by collinhundley** [comment]
Note: changing the `Char` typealias from `Int8` to `Int32` appears to fix the problem. I'd submit a PR but I've modified my fork for MariaDB compatibility. Which I'd love to talk about by the way, but that's another topic.

**Comment by collinhundley** [comment]
UPDATE: Sorry to keep pestering about this, but I believe it's a big flaw in this library that should be addressed. I had a little more time to dig into it today, and I believe that the solution is simple.

My previous comment about changing `Char` to `Int32` was incorrect. Instead, I believe that `Char` should be a typealias for `UInt8`. The `utf8` property of Swift's `String` returns a `UTF8View`, which itself is an array of unsigned 8-bit integers. The documentation isn't clear about this, but after some experimentation I've come to the conclusion that the integers are indeed unsigned, which is why we were experiencing crashes when handling any Unicode characters containing a byte larger than 127.

Making this change allows the package to be fully compatible with all Unicode characters.

**Comment by collinhundley** [comment]
Fixed with PR #27
